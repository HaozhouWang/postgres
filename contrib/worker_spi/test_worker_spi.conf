shared_preload_libraries = 'worker_spi'
worker_spi.monitor_databases = 'contrib_regression'
worker_spi.naptime = 2
