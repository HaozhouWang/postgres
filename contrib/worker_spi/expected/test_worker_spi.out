create extension worker_spi;
select pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

\! pg_ctl -D /tmp/pg_worker_spi_test/data reload
server signaled
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- Test schema quota 
create schema s1;
select diskquota.set_schema_quota_limit('s1', '1 MB');
 set_schema_quota_limit 
------------------------
 
(1 row)

set search_path to s1;
create table a(i int);
insert into a select generate_series(1,100);
insert into a select generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:s1
insert into a select generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:s1
reset search_path;
-- Test role quota
CREATE role u1 NOLOGIN;
CREATE TABLE b (t text);
ALTER TABLE b OWNER TO u1;
select diskquota.set_role_quota_limit('u1', '1 MB');
 set_role_quota_limit 
----------------------
 
(1 row)

insert into b select generate_series(1,100);
insert into b select generate_series(1,100000000);
ERROR:  role's disk space quota exceeded with name:u1
insert into b select generate_series(1,100);
ERROR:  role's disk space quota exceeded with name:u1
drop extension worker_spi;
